src/framework/dockwindow/internal/dockbase.cpp

```diff
@@ -346,6 +361,30 @@ void DockBase::setSeparatorsVisible(bool visible)
     emit separatorsVisibleChanged();
 }
 
+void DockBase::setIsCompact(bool compact)
+{
+    if (m_isCompact == compact) {
+        return;
+    }
+
+    if (compact) {
+        m_nonCompactWidth = width();
```

Comment:
Caching this width and then using it to determine whether the dock can be "uncompacted" may not always be valid, because it assumes that the non-compact width is constant, which may not be the case. For example, 
1. dock is not compact and has width x
2. window becomes smaller and dock becomes compact
3. something else happens that causes the non-compact width to become bigger
4. window is resized to original width; the system thinks the dock can now be uncompacted, because according to cached non-compact width, that would fit; but in reality, that doesn't fit now, because the actual non-compact is now bigger than the cached non-compact width. 

Anyway, if this doesn't cause problems yet in practice, then it's probably good enough for now